pkg/http/listener.go

```diff
@@ -28,43 +28,32 @@ import (
 )
 
 var sslRequiredErrMsg = []byte("HTTP/1.0 403 Forbidden\r\n\r\nSSL required")
+var httpRequiredErrMsg = []byte("HTTP/1.0 403 Forbidden\r\n\r\nPlain-Text required")
+
+// The value chosen below is longest word chosen
+// from all the http verbs comprising of
+// "PRI", "OPTIONS", "GET", "HEAD", "POST",
+// "PUT", "DELETE", "TRACE", "CONNECT".
+const (
+	maxHTTPMethodLen = 7
+)
+
+// HTTP2 PRI method.
+var httpMethodPRI = "PRI"
+
+var defaultHTTP2Methods = []string{
+	httpMethodPRI,
+}
 
-// HTTP methods.
-var methods = []string{
+var defaultHTTP1Methods = []string{
+	http.MethodOptions,
 	http.MethodGet,
 	http.MethodHead,
 	http.MethodPost,
 	http.MethodPut,
-	http.MethodPatch,
 	http.MethodDelete,
-	http.MethodConnect,
-	http.MethodOptions,
 	http.MethodTrace,
-	"PRI", // HTTP 2 method
-}
-
-// maximum length of above methods + one space.
-var methodMaxLen = getMethodMaxLen() + 1
-
-func getMethodMaxLen() int {
-	maxLen := 0
-	for _, method := range methods {
-		if len(method) > maxLen {
-			maxLen = len(method)
-		}
-	}
-
-	return maxLen
-}
-
-func isHTTPMethod(s string) bool {
-	for _, method := range methods {
-		if s == method {
-			return true
-		}
-	}
-
-	return false
+	http.MethodConnect,
```

Comment:
Why have you removed other http methods? Not that this package may be used outside of minio